ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -147,6 +147,11 @@ - (void)updateViewControllerIfNeeded
   // if nav is nil, it means we can be in a fullScreen modal, so there is no nextVC, but we still want to update
   if (vc != nil && (nextVC == vc || isInFullScreenModal || isPresentingVC)) {
     [RNSScreenStackHeaderConfig updateViewController:self.screenView.controller withConfig:self animated:YES];
+    // As the header might have change in `updateViewController` we need to ensure that header height
+    // returned by the `onHeaderHeightChange` event is correct.
+    if ([vc isKindOfClass:[RNSScreen class]]) {
+      [(RNSScreen *)self.screenView.controller calculateAndNotifyHeaderHeightChangeIsModal:NO];
```

Comment:
```suggestion
      [self.screenView.controller calculateAndNotifyHeaderHeightChangeIsModal:NO];
```

This cast should not be necessary as `controller` property on screen returns appropriate type.

Reply:
Changed that in https://github.com/software-mansion/react-native-screens/pull/1917/commits/8884bef22a406281739146dce0ecb4985c482348.